app/Platform/Actions/ComputeGameSortTitleAction.php

```diff
@@ -26,15 +26,45 @@ public function execute(string $gameTitle): string
      *
      * eg: "Mega Man 10" should not be sorted before "Mega Man 2". With the sort titles
      * set to "mega man 00010" and "mega man 00002" respectively, we can mitigate this.
+     *
+     * Standalone numbers with leading zeroes are padded to 7 digits to preserve significance.
+     * eg: "007" -> "0000007"
+     *
+     * Numbers already padded by replaceRomanNumerals (5 digits) are not given more padding.
+     * eg: "00002" remains "00002"
+     *
+     * Numbers attached to letters are not padded.
+     * e.g: "20Q" -> "20Q", "64th" -> "64th"
      */
     private function padNumbers(string $title): string
     {
         return preg_replace_callback(
             // Match numbers not directly attached to letters.
             // In other words, don't convert "Mega Man X4" to "Mega Man X00004".
-            '/(?<![a-zA-Z])(\d+)(?![a-zA-Z])/',
+            '/\b\d+\b/u',
             function ($matches) {
-                return str_pad($matches[0], 5, '0', STR_PAD_LEFT);
+                $number = $matches[0];
+
+                // Check if the number has leading zeroes.
+                if (strlen($number) > 1 && strpos($number, '0') === 0) {
+                    // If the number is already 5 digits (from replaceRomanNumerals), don't pad.
+                    if (strlen($number) === 5) {
+                        return $number;
+                    }
+
+                    // Else, pad to 7 digits to preserve leading zeroes.
+                    if (strlen($number) < 7) {
+                        return str_pad($number, 7, '0', STR_PAD_LEFT);
+                    }
```

Comment:
As the only significant 0s in the database seem to all belong to `007`, this is probably fine. But for a generic solution, I'd recommend padding the number to 5 digits, then inserting the significant zeroes in front of that.

`007` (pad to 5) => `00007` (insert significant zeroes) => `00` + `00007` => `0000007` (result)

Reply:
Adjusted in latest - the test case for 007 is still passing. I believe my change should now properly account for some level of dynamism with significant zeroes.